### ISSUE TITLE
Programmatic API like Istanbul

### ISSUE THREAD

**Issue by manideepak5298** [istanbuljs/nyc#1106]
We are using Istanbul APIs for reporting coverage during run time ( In simpler terms, users will select the folder where js,html files reside. Our tool will generate the code coverage and accumulate them and show it using Istanbul APIs.). Is it possible to have similar APIs like instrumentSync() in Istanbul? Or is there any already existing framework which does it?

**Comment by coreyfarrell** [comment]
nyc is just an application, it does not provide any API's.  API's are available from the monorepo https://github.com/istanbuljs/istanbuljs.  Please note that `istanbul-api` itself is in the process of being deprecated (the git repo shows this but npm is not yet updated).

**Comment by manideepak5298** [comment]
Thank you @coreyfarrell
